src/MarlinSimulator/application.cpp

```diff
@@ -72,67 +75,157 @@ Application::Application() {
+        if (ImPlot::BeginPlot("##Scrolling", "Time (ns)", NULL, ImVec2(-1,150), ImPlotAxisFlags_NoTickLabels | ImPlotFlags_Query, rt_axis, rt_axis)) {
+          ImPlot::PlotLine("pin", &sdata.Data[0].x, &sdata.Data[0].y, sdata.Data.size(), sdata.Offset, sizeof(ImPlotPoint));
+          ImPlot::EndPlot();
         }
-        if (p.pin == monitor_pin) ImGui::SetItemDefaultFocus();
       }
-      ImGui::EndCombo();
-    }
 
-    if (Gpio::pin_map[monitor_pin].event_log.size()) {
-      ScrollingData sdata;
+      static bool export_single_pin = false;
+      if (ImGui::Button("Export selected pin to file")) {
+        export_single_pin = true;
+        ImGuiFileDialog::Instance()->OpenDialog("PulseExportDlgKey", "Choose File", "Value Change Dump (*.vcd){.vcd},.*", ".");
+      }
+
 
-      pin_log_data last{};
-      for (auto v : Gpio::pin_map[monitor_pin].event_log) {
-        if (last.timestamp) sdata.AddPoint(v.timestamp, last.value);
-        sdata.AddPoint(v.timestamp, v.value);
-        last = v;
+      if (ImGui::Button("Export pins matching regex to file")) {
+        export_single_pin = false;
+        ImGuiFileDialog::Instance()->OpenDialog("PulseExportDlgKey", "Choose File", "Value Change Dump (*.vcd){.vcd},.*", ".");
+      }
+
+      static char export_regex[128] = "";
+      ImGui::SameLine();
+      ImGui::InputText("Pin regex", export_regex, sizeof(export_regex));
+
+      if (ImGuiFileDialog::Instance()->Display("PulseExportDlgKey", ImGuiWindowFlags_NoDocking))  {
+        try{
+        if (ImGuiFileDialog::Instance()->IsOk()) {
+          std::string image_filename = ImGuiFileDialog::Instance()->GetFilePathName();
+
+          using namespace vcd;
+
+          HeadPtr head = makeVCDHeader(static_cast<TimeScale>(50), TimeScaleUnit::ns, utils::now());
+          VCDWriter writer{image_filename, std::move(head)};
+
+          if (export_single_pin) {
+            std::string pin_name(active_label);
+            auto scope = pin_name.substr(0, pin_name.find_first_of('_'));
+            auto var = writer.register_var(scope, pin_name, VariableType::wire, 1);
+
+            if (Gpio::pin_map[monitor_pin].event_log.size() && pin_array[monitor_pin].is_digital) {
+              for (const auto &value : Gpio::pin_map[monitor_pin].event_log) {
+                writer.change(var, value.timestamp / 50, utils::format("%u", value.value));
+              }
+            }
+          }
+          else {
+            std::map<size_t, VarPtr> pin_to_var_map;
+            std::regex expression(export_regex);
+
+            for (auto pin : pin_array) {
+              std::string pin_name(pin.name);
+              bool regex_match = strlen(export_regex) == 0 || std::regex_search(pin_name, expression);
+              auto scope = pin_name.substr(0, pin_name.find_first_of('_'));
+              if (pin.is_digital && regex_match) // && Gpio::pin_map[pin.pin].event_log.size() > 1) // Questionable attempt to filter output for inactive lines
```

Comment:
```suggestion
              if (pin.is_digital && regex_match)
```

This was an earlier attempt to reduce output size, before I implemented a regex. Selecting on a pin-by-pin basis might make more sense long-term, but for now the regex was easy to implement.